test/test_io.py

```diff
@@ -138,13 +140,32 @@ def test_read_timestamps_from_packet(self):
             stream = container.streams[0]
             # make sure we went through the optimized codepath
             self.assertIn(b'Lavc', stream.codec_context.extradata)
-            pts_step = int(round(float(1 / (stream.average_rate * stream.time_base))))
-            num_frames = int(round(float(stream.average_rate * stream.time_base * stream.duration)))
-            expected_pts = [i * pts_step for i in range(num_frames)]
 
+            num_frames = int(round(float(stream.average_rate * stream.time_base * stream.duration)))
+            # pts is a global fraction of a second, thus we expect
+            # to see num_frames frames whose global pts is 1/fps
+            expected_pts = [Fraction(i, stream.average_rate) for i in range(num_frames)]
             self.assertEqual(pts, expected_pts)
 
     # TODO add tests for audio
+    def test_audio_cutting(self):
+        with get_tmp_dir() as temp_dir:
+            name = "hmdb51_Turnk_r_Pippi_Michel_cartwheel_f_cm_np2_le_med_6.avi"
+            f_name = os.path.join(temp_dir, name)
+            url = "https://download.pytorch.org/vision_tests/io/" + name
+            try:
+                utils.download_url(url, temp_dir)
+                pts, fps = io.read_video_timestamps(f_name)
+                self.assertEqual(pts, sorted(pts))
+                self.assertEqual(fps, 30)
+            except URLError:
+                msg = "could not download test file '{}'".format(url)
+                warnings.warn(msg, RuntimeWarning)
+                raise unittest.SkipTest(msg)
+
+            lv, la, info = io.read_video(f_name, pts[3], pts[7])
+            # FIXME: add Another video - this one doesn't have audio
```

Comment:
Another option for getting the video would be to use `av.datasets` https://github.com/mikeboers/PyAV/blob/cd458ffe89988b0feca44da6c56ef29f17555962/tests/common.py#L11

Reply:
Sounds reasonable  - I'll take a look.